### PR TITLE
=cache remove `asScala` usage that fails in doc generation

### PR DESCRIPTION
**Pull request by jrudolph**

### CODE REVIEW

**Comment by raboof** [comment]
How can I reproduce this failing doc generation? `sbt -Dakka.genjavadoc.enabled=true javaunidoc:doc` and `sbt clean paradox` on master didn't fail for me

**Comment by jrudolph** [comment]
It was `scalaunidoc:doc` that failed before.

File: akka-http-caching/src/main/scala/akka/http/caching/javadsl/CachingSettings.scala [hunk 1]
```diff
@@ -21,7 +20,7 @@ abstract class CachingSettings private[http] () { self: CachingSettingsImpl ⇒
   def lfuCacheSettings: LfuCacheSettings
 
   def withLfuCacheSettings(newSettings: LfuCacheSettings): CachingSettings =
-    self.copy(lfuCacheSettings = newSettings.asScala)
+    self.copy(lfuCacheSettings = newSettings.asInstanceOf[LfuCacheSettingsImpl]) // crashes with `asScala` in doc generation
```
Comment (raboof) [inline akka-http-caching/src/main/scala/akka/http/caching/javadsl/CachingSettings.scala]: I think we've seen this before - another solution is to move the two imports inside this function, so:
```
  def withLfuCacheSettings(newSettings: LfuCacheSettings): CachingSettings = {
    import akka.http.impl.util.JavaMapping.Implicits._
    import akka.http.caching.CacheJavaMapping.Implicits._

    self.copy(lfuCacheSettings = newSettings.asScala)
  }
```
Just checked, this works and is perhaps slightly neater. Weird though.

Comment (jrudolph) [reply akka-http-caching/src/main/scala/akka/http/caching/javadsl/CachingSettings.scala]: Nice, I had a try to cleanup and simplify JavaMapping a bit in the last days, and I found also lots of weird issues. It's a bit crazy how you stop trying to debug those implicit issues at some point and just try around until it works...

Comment (jrudolph) [reply akka-http-caching/src/main/scala/akka/http/caching/javadsl/CachingSettings.scala]: Will change it to your suggestions.

**Pull request merged**